runtimes/runtimes/operational-telemetry/aws-metric-exporter.ts

```diff
@@ -0,0 +1,172 @@
+            await this.sendOperationalMetrics(operationalMetrics)
+            resultCallback({ code: ExportResultCode.SUCCESS })
+        } catch (error) {
+            console.log('Failed to export metrics', error)
+            resultCallback({ code: ExportResultCode.FAILED })
+            return
+        }
+    }
+
+    forceFlush(): Promise<void> {
+        return Promise.resolve()
+    }
+
+    shutdown(): Promise<void> {
+        this.isShutdown = true
+        return Promise.resolve()
+    }
+
+    async refreshCognitoCredentials(region: string, poolId: string): Promise<Credentials> {
+        if (this.credentials && this.credentialsLastFetched) {
+            const now = new Date()
+            const timeSinceLastFetch = now.getTime() - this.credentialsLastFetched.getTime()
+
+            if (timeSinceLastFetch < this.CREDENTIALS_EXPIRATION_TIME_MS - this.CREDENTIALS_BUFFER_TIME_MS) {
+                return this.credentials
+            }
+        }
+
+        const cognitoIdentity = new CognitoIdentityClient({ region })
+        const identityResponse = await cognitoIdentity.send(
+            new GetIdCommand({
+                IdentityPoolId: poolId,
+            })
+        )
+
+        const credentialsResponse = await cognitoIdentity.send(
+            new GetCredentialsForIdentityCommand({
+                IdentityId: identityResponse.IdentityId,
+            })
+        )
+
+        if (!credentialsResponse.Credentials) {
+            throw new Error('No credentials received')
+        }
+
+        this.credentialsLastFetched = new Date()
+        this.credentials = credentialsResponse.Credentials
+
+        return this.credentials
+    }
+
+    signRequest(url: URL, body: string, region: string, credentials: Credentials) {
+        const request = new HttpRequest({
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+                host: url.host,
+            },
+            hostname: url.host,
+            path: url.pathname,
+            body: body,
+        })
+
+        const signer = new SignatureV4({
+            credentials: {
+                accessKeyId: credentials.AccessKeyId!,
+                secretAccessKey: credentials.SecretKey!,
+                sessionToken: credentials.SessionToken!,
+            },
+            region: region,
+            service: 'execute-api',
+            sha256: Sha256,
+        })
+
+        return signer.sign(request)
+    }
+
+    extractOperationalMetrics(metrics: ResourceMetrics): OperationalMetric[] {
+        return metrics.scopeMetrics
+            .map((scopeMetrics: ScopeMetrics) => {
+                return scopeMetrics.metrics.map((metric: MetricData) => {
+                    return this.transform(metric, metrics.resource)
+                })
+            })
+            .flat()
+    }
+
+    transform(metric: MetricData, resource: Resource): OperationalMetric {
+        const dataPoint = metric.dataPoints[0]
```

Comment:
Why only the 1st datapoint? What if there are none?

Reply:
still WIP